src/credentials/CHIPCertToX509.cpp

```diff
@@ -650,6 +652,7 @@ static CHIP_ERROR DecodeConvertExtensions(TLVReader & reader, ASN1Writer & write
                 err = DecodeConvertExtension(reader, writer, certData);
                 SuccessOrExit(err);
             }
+            VerifyOrExit(err == CHIP_END_OF_TLV, /* returns err */);
```

Comment:
This pattern looks really unclean. We should not need to have a "pure text repalcement" where it looks we have a missing argument.